internal/xds/rbac/converter_test.go

```diff
@@ -110,5 +112,60 @@ func (s) TestBuildLoggerErrors(t *testing.T) {
 
 		})
 	}
+}
+
+func (s) TestBuildLoggerKnownTypes(t *testing.T) {
+	tests := []struct {
+		name         string
+		loggerConfig *v3rbacpb.RBAC_AuditLoggingOptions_AuditLoggerConfig
+	}{
+		{
+			name: "stdout logger",
+			loggerConfig: &v3rbacpb.RBAC_AuditLoggingOptions_AuditLoggerConfig{
+				AuditLogger: &v3corepb.TypedExtensionConfig{
+					Name:        stdout.Name,
+					TypedConfig: createStdoutPb(t),
+				},
+				IsOptional: false,
+			},
+		},
+		{
+			name: "stdout logger with generic TypedConfig",
+			loggerConfig: &v3rbacpb.RBAC_AuditLoggingOptions_AuditLoggerConfig{
+				AuditLogger: &v3corepb.TypedExtensionConfig{
+					Name:        stdout.Name,
+					TypedConfig: createXDSTypedStruct(t, map[string]interface{}{}, stdout.Name),
+				},
+				IsOptional: false,
+			},
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			logger, err := buildLogger(test.loggerConfig)
+			if err != nil {
+				t.Fatalf("expected success. got error: %v", err)
+			} else {
+				switch test.name {
+				case "stdout logger":
```

Comment:
I don't think switching behavior on the testcase name is proper.  The name is there for descriptive purposes only.  Everything else should be covered by a parameter in the testcase struct.

In this case, if you want to test that the proper logger was built, you should be able to find the logger's type by looking it up in the registry, too, instead of exporting it.

Is this not something that should be tested for the other test case, too?

Reply:
Yes it should be tested for the other test case, only keeping it out was a holdover from when I had it combined with the other test in this file.
Will clean it up

Reply:
I'm not 100% clear what you mean by getting the correct logger type from the registry.

But that's also something we can kick down the road because there's only one logger type right now, so I can leave it hard-coded in as it is right now if that is okay with you

Reply:
PTAL